modules/rosetta/src/main/scala/org/tessellation/rosetta/domain/Amount.scala

```diff
@@ -1,31 +1,54 @@
 package org.tessellation.rosetta.domain
 
 import org.tessellation.ext.derevo.magnoliaCustomizable.snakeCaseConfiguration
-import org.tessellation.rosetta.domain.currency.Currency
+import org.tessellation.rosetta.domain.currency.{Currency, DAG}
+import org.tessellation.schema.transaction.{TransactionAmount, TransactionFee}
 
-import derevo.cats.eqv
+import derevo.cats.{eqv, show}
 import derevo.circe.magnolia._
 import derevo.derive
 import eu.timepit.refined.api.Refined
 import eu.timepit.refined.auto._
 import eu.timepit.refined.boolean.Not
 import eu.timepit.refined.cats._
 import eu.timepit.refined.generic.Equal
+import eu.timepit.refined.numeric.{NonNegative, Positive}
+import eu.timepit.refined.refineV
 import io.circe.refined._
 import io.estatico.newtype.macros.newtype
 
 object amount {
-  @derive(eqv, customizableDecoder, customizableEncoder)
+  @derive(eqv, customizableDecoder, customizableEncoder, show)
   case class Amount(
     value: AmountValue,
     currency: Currency
-  )
+  ) {
+    def negate: Either[String, Amount] =
+      value.negate.map(Amount(_, currency))
+  }
 
-  type AmountValueRefined = Long Refined Not[Equal[0L]]
+  type AmountValuePredicate = Not[Equal[0L]]
+  type AmountValueRefined = Long Refined AmountValuePredicate
 
-  @derive(eqv, decoder, encoder)
+  @derive(eqv, decoder, encoder, show)
   @newtype
   case class AmountValue(value: AmountValueRefined) {
     def isNegative: Boolean = value < 0L
+    def isPositive: Boolean = value > 0L
+
+    def negate: Either[String, AmountValue] =
+      refineV[AmountValuePredicate](-value.value).map(rv => AmountValue(rv))
+
+    def toTransactionAmount: Either[String, TransactionAmount] =
+      refineV[Positive](value.value).map(TransactionAmount(_))
+
+    def toTransactionFee: Either[String, TransactionFee] =
+      refineV[NonNegative](value.value).map(TransactionFee(_))
+  }
+
+  object Amount {
```

Comment:
a rule of thumb is that corresponding object should be next to the case class declaration. otherwise it's harder to read/reason about

```
case class Amount ...

object Amount ....

...
```